hashes/src/hkdf/mod.rs

```diff
@@ -32,6 +32,7 @@ impl fmt::Display for MaxLengthError {
 impl std::error::Error for MaxLengthError {}
 
 /// HMAC-based Extract-and-Expand Key Derivation Function (HKDF).
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
```

Comment:
I'm afraid `Debug` leaks secret material here and if that is the case then the equality/std hashing functions also leak it via timing side channel.

I suggest we derive `Copy` and `Clone` and implement the rest except `*Ord` by running it through some cryptographic hash function.

Reply:
Thanks man. You correctly led me to see that I did not grab the correct set of derives for hash types. The list on other hash types is 

```rust
#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
```

As for the other comments about side channel attacks since `internal_macros` and `hmac` have these same set of traits I believe it is out of scope for this PR.

Reply:
> The list on other hash types is

Presumably because `Debug` is implemented manually to output hex rather than slices but derive "could work" here except it doesn't because it'd be outputting secret.

> since `internal_macros` and `hmac` have these same set of traits

How they are related? The *output* of hmac is not secret (usually) but when used as input in KDF it is.

> I believe it is out of scope for this PR.

Then remove the offending traits for security reasons. Let's not add broken stuff (that is frankly useless, HKDF is more like RNG rather than a value) We can add them later if we wish.

Reply:
+1 to just removing them for now (partialeq, eq, partialord, ord). I would like to eventually have equality but we'll need to be careful about it and it's nonbreaking to add it later.

I could go either way on debug leaking secret data. Since we already did the hash thing for some other secret types we might as well do it here.